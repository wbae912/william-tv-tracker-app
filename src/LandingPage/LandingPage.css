.landing-page {
  box-sizing: border-box;
  font-family: 'Montserrat', sans-serif;
  background-color: rgb(43, 38, 38);
  color: white;
  padding-top: 65px;
}

main {
  text-align: center;
  border: 1px solid black;
}

.header-tag {
  font-weight: bold;
  font-size: 30px;
}

.underline {
  margin-left: 1.5%;
  margin-right: 1.5%;
}

.section-p {
  margin-right: 2%;
  margin-left: 2%;
  font-size: 13px;
}