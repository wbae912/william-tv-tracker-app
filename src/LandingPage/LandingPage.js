import React, { Component } from 'react';

export default class LandingPage extends Component {
  render() {
    return (
    <>
      <nav className="nav">
        <h2 className="nav-app-title">TV Tracker</h2>
        <div className="landing-page-links">
          <button type="button" className="login-button">Login</button>
          <button type="button" className="register-button">Register</button>
        </div>
      </nav>
      <main>
        <header className="header">
          <h1 className="app-title">TV Tracker</h1>
          <p className="app-title-headline">track and catalog your favorite TV shows...</p>
        </header>

        <hr className="underline"></hr>

        <section className="section-one">
          <h2 className="section-h2">Create Your TV Show Entries</h2>
          <p className="section-p">TV Tracker allows you to submit TV shows that you are planning to watch, currently watching, or completed! Did a friend tell 
            you about a good show to watch? Add it as an entry as a reminder to see later.</p>
        </section>

        <hr className="underline"></hr>

        <section className="section-two">
          <h2 className="section-h2">Keep track of your Shows</h2>
          <p className="section-p">TV Tracker keeps a record of your TV shows so you can always have them on hand. Are you having a hard 
            time remembering which episode you left off for that one show? Keep track of it within the app? Like seeing a list of all the 
            TV shows you've watched? Go through your list and recommend one to a friend!</p>
        </section>

        <hr className="underline"></hr>

        <section className="section-three">
          <h2 className="section-h2">View Your Dashboard</h2>
          <p className="section-p">TV Tracker provides a convenient dashboard that displays useful and easily digestible information for 
            you. Want to see how many TV shows you are currently watching? Curious about how many shows you rated 5 stars? Or, perhaps 
            you want to see what your favorite TV show genre is? The dashboard is there for you so it takes all the leg work out of having 
            to find out for yourself!</p>
        </section>

        <hr className="underline"></hr>

        <section className="register-section">
          <h2 className="section-h2">Start Today!</h2>
          <form className="register-form">
            <div className="register-form-div">
              <label htmlFor="first-name">First Name</label>
              <input type="text" id="first-name"/>
            </div>
            <div className="register-form-div">
              <label htmlFor="last-name">Last Name</label>
              <input type="text" id="last-name"/>
            </div>
            <div className="register-form-div">
              <label htmlFor="email">Email</label>
              <input type="email" id="email" placeholder="email@example.com"/>
            </div>
            <div className="register-form-div">
              <label htmlFor="password">Password</label>
              <input type="text" id="password"/>
            </div>
            <button type="submit" className="reigster-button">Sign Up</button>
          </form>
        </section>
      </main>
    </>
    )
  }
}