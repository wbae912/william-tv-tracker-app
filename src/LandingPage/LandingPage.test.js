import React from 'react';
import ReactDOM from 'react-dom';
import LandingPage from './LandingPage';
import { MemoryRouter } from 'react-router';
import renderer from 'react-test-renderer';

describe('LandingPage component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<MemoryRouter><LandingPage /></MemoryRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  
  it('renders UI as expected', () => {
    const tree = renderer
      .create(<MemoryRouter><LandingPage /></MemoryRouter>)
      .toJSON();
      expect(tree).toMatchSnapshot();
  })
})