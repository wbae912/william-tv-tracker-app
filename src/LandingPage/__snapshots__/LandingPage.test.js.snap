// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage component renders UI as expected 1`] = `
<div
  className="landing-page"
>
  <main
    role="main"
  >
    <header
      className="header"
    >
      <h1
        className="header-tag"
      >
        Explore TV Tracker
      </h1>
      <div
        className="hero"
      >
        <h2
          className="app-title-headline"
        >
          Track and catalog your favorite TV shows...
        </h2>
      </div>
    </header>
    <div
      className="section-flex"
    >
      <section
        className="section-one"
      >
        <h2
          className="section-h2"
        >
          Add Your TV Shows
        </h2>
        <i
          className="fa fa-plus fa-2x"
        />
        <p
          className="section-p"
        >
          Did a friend recommend you a good show to watch? Add it as an entry so you don't forget.
        </p>
      </section>
      <hr
        className="underline"
      />
      <section
        className="section-two"
      >
        <h2
          className="section-h2"
        >
          Track Your Shows
        </h2>
        <i
          className="fa fa-eye fa-2x"
        />
        <p
          className="section-p"
        >
          Struggling to remember which episode you left off on? TV Tracker saves your entries and organizes them into convenient lists. Take control by editing and deleting entries in your lists.
        </p>
      </section>
      <hr
        className="underline"
      />
      <section
        className="section-three"
      >
        <h2
          className="section-h2"
        >
          View Your Dashboard
        </h2>
        <i
          className="fa fa-list fa-2x"
        />
        <p
          className="section-p"
        >
          Curious about how many shows you rated 5 stars? TV Tracker's built-in dashboard provides a quick snapshot view of all the important metrics.
        </p>
      </section>
    </div>
  </main>
</div>
`;
